include/dca/phys/dca_step/cluster_mapping/update_chemical_potential.hpp

```diff
@@ -32,6 +32,8 @@
 #include "dca/phys/domains/cluster/cluster_domain_aliases.hpp"
 #include "dca/util/print_time.hpp"
 
+#undef B0
```

Comment:
This needs a comment. A clear one

Reply:
`B0` is a macro defined in a Summit system file as well as a variable used in DCA++, which causes conflict. So I have to undef B0 in this file, otherwise, cause compilation error. So, on Summit, there is a file `/usr/include/bits/termios.h`, which is included by boost library (either Summit module or you-built-your-own), which then included by HPX, which then further appeared in HPX-enabled DCA++. In that file,
```
#define  B0     0000000         /* hang up */
```
The `B0` variable is also used in DCA++, here: https://github.com/CompFUSE/DCA/blob/2d2e8e4ef2c49446d01580db39954f536202d61b/include/dca/phys/dca_step/cluster_mapping/update_chemical_potential.hpp#L271

If not under B0, the following error occurs.
```
In file included from /usr/include/termios.h:40,
                 from /usr/include/bits/ioctl-types.h:5,
                 from /usr/include/sys/ioctl.h:29,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/detail/socket_types.hpp:59,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/detail/epoll_reactor.hpp:29,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/detail/reactor.hpp:21,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/detail/impl/scheduler.ipp:23,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/detail/scheduler.hpp:223,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/system_context.hpp:19,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/impl/system_executor.hpp:22,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/system_executor.hpp:129,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/associated_executor.hpp:21,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/detail/bind_handler.hpp:20,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/detail/wrapped_handler.hpp:18,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/io_context.hpp:24,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/boost-1.70.0-dcad7jn6fxdv62u2dsx4lunl57ndelc4/include/boost/asio/io_service.hpp:18,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/hpx-1.4.1-63vpmjgrdjy45axclzwizucfaa7am5nr/include/hpx/util/io_service_pool.hpp:19,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/hpx-1.4.1-63vpmjgrdjy45axclzwizucfaa7am5nr/include/hpx/threadmanager.hpp:23,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/hpx-1.4.1-63vpmjgrdjy45axclzwizucfaa7am5nr/include/hpx/runtime/applier/applier.hpp:22,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/hpx-1.4.1-63vpmjgrdjy45axclzwizucfaa7am5nr/include/hpx/runtime/components/server/fixed_component_base.hpp:13,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/hpx-1.4.1-63vpmjgrdjy45axclzwizucfaa7am5nr/include/hpx/runtime/agas/server/primary_namespace.hpp:20,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/hpx-1.4.1-63vpmjgrdjy45axclzwizucfaa7am5nr/include/hpx/lcos/detail/async_colocated.hpp:18,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/hpx-1.4.1-63vpmjgrdjy45axclzwizucfaa7am5nr/include/hpx/collectives/reduce.hpp:81,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/hpx-1.4.1-63vpmjgrdjy45axclzwizucfaa7am5nr/include/hpx/include/lcos.hpp:21,
                 from /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-8.1.1/hpx-1.4.1-63vpmjgrdjy45axclzwizucfaa7am5nr/include/hpx/hpx.hpp:14,
                 from /gpfs/alpine/proj-shared/cph102/weile/dev/src/hpx_dca/include/dca/parallel/hpx/hpx.hpp:19,
                 from /gpfs/alpine/proj-shared/cph102/weile/dev/src/hpx_dca/build_hpx/include/dca/config/threading.hpp:21,
                 from /gpfs/alpine/proj-shared/cph102/weile/dev/src/hpx_dca/test/system-level/dca/dca_sp_DCA+_thread_test.cpp:19:
/gpfs/alpine/proj-shared/cph102/weile/dev/src/hpx_dca/include/dca/phys/dca_step/cluster_mapping/update_chemical_potential.hpp: In member function 'void dca::phys::clustermapping::update_chemical_potential<parameters_type, MOMS_type, coarsegraining_type>::compute_density_correction(dca::func::function<double, dca::func::dmn_variadic<dca::func::dmn_0<dca::phys::domains::electron_band_domain>, dca::func::dmn_0<dca::phys::domains::electron_spin_domain> > >&)':
/gpfs/alpine/proj-shared/cph102/weile/dev/src/hpx_dca/include/dca/phys/dca_step/cluster_mapping/update_chemical_potential.hpp:273:63: error: expected unqualified-id before numeric constant
   func::function<double, func::dmn_variadic<nu, KClusterDmn>> B0;
                                                               ^~
In file included from /gpfs/alpine/proj-shared/cph102/weile/dev/src/hpx_dca/include/dca/phys/dca_loop/dca_loop.hpp:29,
                 from /gpfs/alpine/proj-shared/cph102/weile/dev/src/hpx_dca/test/system-level/dca/dca_sp_DCA+_thread_test.cpp:28:
/gpfs/alpine/proj-shared/cph102/weile/dev/src/hpx_dca/include/dca/phys/dca_step/cluster_mapping/update_chemical_potential.hpp:289:76: error: expression cannot be used as a function
         std::complex<double> G0 = 1. / (I_wn + A0(nu_i, k_i) + B0(nu_i, k_i) / I_wn);

```

Reply:
That is extremely weird from summit. Still, I think renaming the variable to b0 is more portable.

Reply:
renamed B0 to b0 in new commit.

https://github.com/STEllAR-GROUP/DCA/blob/ecb96fb7ef2991b12a9fe8772fb827453302035d/include/dca/phys/dca_step/cluster_mapping/update_chemical_potential.hpp#L272

Reply:
update_chemical_potential is an nice documentation and variable naming antipattern.